joomla/joomla-cms
# Login URL is wrong (doesn't check layout)

### Summary of Changes
The Legacy router grabs the incorrect item ID if you've created a logout menu item before creating the login menu; that's because it doesn't verify the layout. This patch fixes that. No idea how to fix the Modern routing though.


### Testing Instructions
- Make sure SEF is enabled
- Make sure you're using `URL Routing` set to `Legacy` in `Users > Manage > Options > Integration`
- The order is important: create a new `Users > Logout` menu item and after that, create a new `Users > Login` menu item.
- Run the following code:
```php
require_once JPATH_SITE.'/components/com_users/helpers/route.php';
var_dump(JRoute::_('index.php?option=com_users&view=login'));
var_dump(UsersHelperRoute::getLoginRoute());
die;
```

### Expected result
```
string(21) "/joomla/index.php/login" string(3) "110" 
```


### Actual result
```
string(21) "/joomla/index.php/logout" string(3) "109" 
```

### Documentation Changes Required
